modules/core/src/scenegraph/gltf-instantiator/gltf-material.js

```diff
@@ -46,13 +46,97 @@ const fs = `
   }
 `;
 
+// TODO: Move to new file
+class GLTFEnv {
+  constructor(gl) {
+    this.gl = gl;
+  }
+
+  getTexUrl(envMap, type, dir, mipLevel = 0) {
+    return `https://raw.githubusercontent.com/KhronosGroup/glTF-WebGL-PBR/master/textures/${envMap}/${type}/${type}_${dir}_${mipLevel}.jpg`;
+  }
+
+  getBrdfUrl() {
+    return "https://raw.githubusercontent.com/KhronosGroup/glTF-WebGL-PBR/master/textures/brdfLUT.png";
+  }
+
+  getImageAsync(imageUrl) {
+    /* global Image */
+    return new Promise(resolve => {
+      const img = new Image();
+      img.onload = () => resolve(img);
+      img.crossOrigin = 'anonymous';
+      img.src = imageUrl;
+    });
+  }
+
+  makeCube(id, func) {
```

Comment:
Should we try to get the async textures working. I think we should be able to make this simpler. Wouldn't it be nicer if we just called this with:

```
   new TextureCube(...
        [this.gl.TEXTURE_CUBE_MAP_POSITIVE_X]: loadImage(`${baseUrl}-right...`),
        [this.gl.TEXTURE_CUBE_MAP_POSITIVE_Y]: ...,
        [this.gl.TEXTURE_CUBE_MAP_POSITIVE_Z]: ...,
        [this.gl.TEXTURE_CUBE_MAP_NEGATIVE_X]: ...,
        [this.gl.TEXTURE_CUBE_MAP_NEGATIVE_Y]: loadfaces[4],
        [this.gl.TEXTURE_CUBE_MAP_NEGATIVE_Z]: loadfaces[5]
```

Then we can drop this method.

Reply:
I spend many hours on that and couldn't get it to work. Sorry 😭